utils/test_harness/sysman/src/test_harness_sysman_device.cpp

```diff
@@ -24,12 +24,36 @@ zes_device_properties_t
 get_sysman_device_properties(zes_device_handle_t device) {
   zes_device_properties_t properties = {ZES_STRUCTURE_TYPE_DEVICE_PROPERTIES,
                                         nullptr};
-
   EXPECT_EQ(ZE_RESULT_SUCCESS, zesDeviceGetProperties(device, &properties));
-
   return properties;
 }
 
+std::vector<zes_subdevice_exp_properties_t>
+get_sysman_subdevice_properties(zes_device_handle_t device, uint32_t *count) {
+  std::vector<zes_subdevice_exp_properties_t> sub_device_properties(*count);
```

Comment:
Here you should get the count from zeDeviceGetSubDevicePropertiesExp() first and then create this vector with appropriate size.

Reply:
Yes thanks for pointing out. Have made the calls simpler.